.github/workflows/centos7.yml

```
@@ -21,6 +21,7 @@ env:
 jobs:
 
   doc_generation:
+    if: ${{ github.event_name == 'release' || contains(github.ref_name, 'doc/') }}
```

```suggestion
    if: ${{ github.event_name == 'release' || startsWith(github.ref_name, 'doc/') }}
```
Slightly more accurate